app/post.go

```diff
@@ -67,6 +86,13 @@ func CreatePost(post *model.Post, teamId string, triggerWebhooks bool) (*model.P
 		pchan = Srv.Store.Post().Get(post.RootId)
 	}
 
+	if utils.IsLicensed() && *utils.Cfg.TeamSettings.ExperimentalTownSquareIsReadOnly &&
+		!post.IsSystemMessage() &&
+		!UserHasPermissionToExecute(post.UserId, model.PERMISSION_MANAGE_SYSTEM) &&
```

Comment:
Can you move the call to get the user up from UserHasPermissionToExecute into CreatePost and then pass the user to the handlePostEvents call at the end of this function? Otherwise, we're sometimes getting user object twice when a post is made